atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionPreCommitActions.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import com.google.common.collect.Maps;
+import com.palantir.atlasdb.common.api.timelock.TimestampLeaseName;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.LongSupplier;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+final class TransactionPreCommitActions {
+
+    static class PreCommitAction {
+        final Consumer<LongSupplier> action;
+        final int timestampCount;
+
+        PreCommitAction(Consumer<LongSupplier> action, int timestampCount) {
```

Comment:
As a reader, it's not obvious to me what the role of the long supplier here is (I know from context, but this needs to be comprehensible to readers with less context).